.github/workflows/main.yml

```diff
@@ -2,8 +2,67 @@ name: CI
 on: [push, pull_request]
 
 jobs:
-  test:
-    name: Test
+  buildlibc:
+    name: Build libc
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: true
+
+    - name: Install clang (Windows)
+      shell: bash
+      run: |
+          curl -fsSLO https://releases.llvm.org/8.0.0/LLVM-8.0.0-win64.exe
+          7z x LLVM-8.0.0-win64.exe -y -o"llvm"
+          echo "::add-path::$(pwd)/llvm/bin"
+          echo "::set-env name=WASM_AR::$(pwd)/llvm/bin/llvm-ar.exe"
+      if: matrix.os == 'windows-latest'
+
+    - name: Install llvm-nm (Windows)
+      shell: bash
+      run: |
+        # Add --no-self-update as a workaround for
+        # https://github.com/microsoft/azure-pipelines-image-generation/issues/1224
+        rustup update stable --no-self-update
+        rustup default stable
+        rustup component add llvm-tools-preview
+        echo "::set-env name=WASM_NM::$(rustc --print sysroot|sed 's|C:|/c|'|sed 's|\\|/|g')/lib/rustlib/x86_64-pc-windows-msvc/bin/llvm-nm.exe"
+      if: matrix.os == 'windows-latest'
+
+    - name: Install clang (MacOS)
+      shell: bash
+      run: |
+        curl -sSf http://releases.llvm.org/8.0.0/clang+llvm-8.0.0-x86_64-apple-darwin.tar.xz | tar xJf -
+        export CLANG_DIR=`pwd`/clang+llvm-8.0.0-x86_64-apple-darwin/bin
+        echo "::add-path::$CLANG_DIR"
+        echo "::set-env name=WASM_CC::$CLANG_DIR/clang"
+      if: matrix.os == 'macos-latest'
+
+    - name: Install clang (Linux)
+      shell: bash
+      run: |
+        curl -sSf http://releases.llvm.org/8.0.0/clang+llvm-8.0.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz | tar xJf -
+        export CLANG_DIR=`pwd`/clang+llvm-8.0.0-x86_64-linux-gnu-ubuntu-16.04/bin
+        echo "::add-path::$CLANG_DIR"
+        echo "::set-env name=WASM_CC::$CLANG_DIR/clang"
+      if: matrix.os == 'ubuntu-latest'
+
+    - name: Build libc
+      shell: bash
+      run: make -j4
+
+    - uses: actions/upload-artifact@v1
+      with:
+        # Upload the sysroot folder. Give it a name according to the OS it was built for.
+        name: ${{ format( 'sysroot-{0}.tgz', matrix.os) }}
+        path: sysroot
```

Comment:
Hmm, do we need this?

Firstly, it should be platform-independent so at least I would upload only from the linux builder.  

Secondly I wonder if we should call it something other than sysroot (maybe wasi-libc-prebuilt?).

Thirdly, does any actually need/want this artifact.   I'm not sure that very useful on its own without the rest of wasi-sdk (wasi-sdk already uploads its own complete sysroot for those who have their own clang).

Reply:
I think this is useful only for troubleshooting builds. It doesn't cost anything (well, it probably costs Microsoft something in storage, but they have the money) or publish the artifacts anywhere prominent, but authors and reviewers can use it to verify that the sysroot built what they expected on platforms they don't have access to.

Ideally we'd add a separate step that downloads all three artifacts and verifies that they're the same, since we expect the build to be the same on all operating systems.

We do the same thing in wasi-sdk PRs as well, for the same reasons.

Reply:
I don't mind uploading for CI/debugging purposes.  I was just thinking this might not make sense in the github releases.. is that a separate thing?

Reply:
yeah, this is a separate thing from github releases. Agree that its confusing, I had to figure it out the hard way too